### ISSUE TITLE
Semantics `headingLevel`doc update

### ISSUE THREAD

**Issue by RamonFarizel** [flutter/flutter#179998]
[headingLevel](https://api.flutter.dev/flutter/semantics/SemanticsProperties/headingLevel.html) should be updated to indicate that it also affects Android.

This relates to issue https://github.com/flutter/flutter/issues/179678, which reported that the `header` property (which sets `Flag.IS_HEADER`) no longer works on Android after the code was changed to only check `headingLevel > 0` instead of also checking `IS_HEADER`.
